src/extension.ts

```diff
@@ -57,8 +57,12 @@ export function activate(context: vscode.ExtensionContext) {
 	context.subscriptions.push(azureAccount.onSessionsChanged(() => explorer.refresh()));
 	context.subscriptions.push(vscode.window.registerTreeDataProvider('cosmosDBExplorer', explorer));
 
-	const documentEditor: DocumentEditor = new DocumentEditor(context);
-	context.subscriptions.push(documentEditor);
+	const documentEditor: DocumentEditor = new DocumentEditor(context, 'cosmos-document.json');
+	const collectionEditor: DocumentEditor = new DocumentEditor(context, 'cosmos-collection.json');
+	const resultEditor: DocumentEditor = new DocumentEditor(context, 'cosmos-result.json');
```

Comment:
Have precisely one file for the resultEditor. 
Regression from previous implementation - can no longer have multiple result.json files which can be persisted to azure.

Reply:
I still think there should be only one DocumentEditor. However, inside of DocumentEditor, there should be a map of:
filepath -> CollectionNode | DocumentNode

If the filePath is a new temporary folder each time, then we don't have to limit the user to one result.json file